File: core/src/main/java/org/maibot/core/event/SystemEventService.java
package org.maibot.core.event;

import io.netty.channel.Channel;
import io.netty.channel.ChannelHandler;
import io.netty.channel.embedded.EmbeddedChannel;
import org.maibot.sdk.ioc.Component;
import org.maibot.sdk.ioc.DestroyableComponent;
import org.slf4j.LoggerFactory;

@Component
public class SystemEventService implements DestroyableComponent {
    private static final org.slf4j.Logger log = LoggerFactory.getLogger(SystemEventService.class);
    private final        Channel          channel;

    public SystemEventService() {
        this.channel = new EmbeddedChannel();
    }

    public void addHandler(String name, ChannelHandler handler) {
        this.channel.pipeline().addLast(name, handler);
    }

    public void removeHandler(String name) {
        this.channel.pipeline().remove(name);
    }

    @Override
    public void preDestroy() {
        try {
            this.channel.close();
        } catch (Exception e) {
            log.error("关闭系统事件服务时发生错误", e);
        }
    }
}

File: core/src/main/java/org/maibot/core/event/SystemEventService.kt
package org.maibot.core.event

import io.netty.channel.Channel
import io.netty.channel.ChannelHandler
import io.netty.channel.embedded.EmbeddedChannel
import org.maibot.sdk.ioc.Component
import org.maibot.sdk.ioc.DestroyableComponent
import org.slf4j.Logger
import org.slf4j.LoggerFactory

@Component
class SystemEventService : DestroyableComponent {
    private val channel: Channel = EmbeddedChannel()

    init {
        log.info("系统事件服务已启动")
    }

    fun addHandler(name: String?, handler: ChannelHandler?) {
        this.channel.pipeline().addLast(name, handler)
    }

    fun removeHandler(name: String?) {
        this.channel.pipeline().remove(name)
    }

    fun fireEvent(event: Any?) {
        this.channel.pipeline().fireUserEventTriggered(event)
    }

    override fun preDestroy() {
        try {
            this.channel.close()
        } catch (e: Exception) {
            log.error("关闭系统事件服务时发生错误", e)
        }
    }

    companion object {
        private val log: Logger = LoggerFactory.getLogger(SystemEventService::class.java)
    }
}
